woocommerce/woocommerce-ios
# [Issue Refund] Adds Coordinating Controller

part of #2844 

# Why

This PR does not add any new functionality. It pays some tech debt by adding a coordinating controller to handle all presentation/navigation related to the issue refund feature. The main goal of this change is:

- Abstract the navigation/presentation duties from the view controllers.
- Make that navigation/presentation testable.

* Tests will be added in the next PR to maintain this one in a manageable size.


# How
- Added `IssueRefundCoordinatingController` that orquestrates `IssueRefundViewController`, `RefundConfirmattionViewController`, `ListSelectorViewController`, `UIAlertViewController`, and  `InProgressViewController` navigation and presentation.

- Refactor `IssueRefundViewController` and `RefundConfirmattionViewController` to expose actions that will be handled by `IssueRefundCoordinatingController`.

- Modify `RefundItemQuantityListSelectorCommand` to include the item index to be used when the view controller returns from the quantity selection.

# GIFS

## Success
![sucess refund](https://user-images.githubusercontent.com/562080/100391699-4889f280-3002-11eb-955b-de1d6ce01456.gif)


## Failure
![refund-error](https://user-images.githubusercontent.com/562080/100391677-327c3200-3002-11eb-8419-a2aae391ebaa.gif)


# Testing steps

- Smoke test creating refunds, with and without internet connection(to see the error notice)

Update release notes:
